lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -4,19 +4,147 @@
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+import uk.gov.di.ipv.core.library.helpers.RequestHelper;
+import uk.gov.di.ipv.core.library.persistence.item.IpvSessionItem;
+import uk.gov.di.ipv.core.library.service.CiStorageService;
+import uk.gov.di.ipv.core.library.service.ConfigurationService;
+import uk.gov.di.ipv.core.library.service.IpvSessionService;
+import uk.gov.di.ipv.core.library.service.UserIdentityService;
+
+import java.text.ParseException;
+import java.util.List;
+import java.util.Optional;
 
 public class CheckExistingIdentityHandler
         implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    public static final List<Gpg45Profile> ACCEPTED_PROFILES =
+            List.of(Gpg45Profile.M1A, Gpg45Profile.M1B);
+    public static final JourneyResponse JOURNEY_REUSE = new JourneyResponse("/journey/reuse");
+    public static final JourneyResponse JOURNEY_NEXT = new JourneyResponse("/journey/next");
+    private static final Logger LOGGER = LogManager.getLogger();
+    private final ConfigurationService configurationService;
+    private final UserIdentityService userIdentityService;
+    private final IpvSessionService ipvSessionService;
+    private final Gpg45ProfileEvaluator gpg45ProfileEvaluator;
+    private final CiStorageService ciStorageService;
+
+    public CheckExistingIdentityHandler(
+            ConfigurationService configurationService,
+            UserIdentityService userIdentityService,
+            IpvSessionService ipvSessionService,
+            Gpg45ProfileEvaluator gpg45ProfileEvaluator,
+            CiStorageService ciStorageService) {
+        this.configurationService = configurationService;
+        this.userIdentityService = userIdentityService;
+        this.ipvSessionService = ipvSessionService;
+        this.gpg45ProfileEvaluator = gpg45ProfileEvaluator;
+        this.ciStorageService = ciStorageService;
+    }
 
     @ExcludeFromGeneratedCoverageReport
-    public CheckExistingIdentityHandler() {}
+    public CheckExistingIdentityHandler() {
+        this.configurationService = new ConfigurationService();
+        this.userIdentityService = new UserIdentityService(configurationService);
+        this.ipvSessionService = new IpvSessionService(configurationService);
+        this.gpg45ProfileEvaluator = new Gpg45ProfileEvaluator(configurationService);
+        this.ciStorageService = new CiStorageService(configurationService);
+    }
 
     @Override
     public APIGatewayProxyResponseEvent handleRequest(
-            APIGatewayProxyRequestEvent input, Context context) {
-        return ApiGatewayResponseGenerator.proxyJsonResponse(HttpStatus.SC_OK, "Hello world");
+            APIGatewayProxyRequestEvent event, Context context) {
+        LogHelper.attachComponentIdToLogs();
+
+        try {
+            String ipvSessionId = RequestHelper.getIpvSessionId(event);
+            LogHelper.attachIpvSessionIdToLogs(ipvSessionId);
```

Comment:
This isn't needed. `RequestHelper.getIpvSessionId(event);` method already attaches the sessionID to the logs

Reply:
Good spot, removed